lib/rake/javaextensiontask.rb

```diff
@@ -108,8 +108,11 @@ def define_compile_tasks(for_platform = nil, ruby_ver = RUBY_VERSION)
         EOF
         warn_once(not_jruby_compile_msg) unless defined?(JRUBY_VERSION)
 
+        javac_path = File.exist?(t = File.join(ENV.fetch("JAVA_HOME", "/"), "bin", "javac")) ?
+          t : "javac"
```

Comment:
Using `/` as the default `JAVA_HOME` will not work when `javac` exists in both of `/usr/bin/javac` and `/usr/local/bin/javac`, and `PATH` is `/usr/local/bin:/usr/bin:/bin`.

We may need to check `ENV["JAVA_HOME"]` explicitly:

```suggestion
        java_home = ENV["JAVA_HOME"]
        if java_home
          javac_path = File.join(java_home, "bin", "javac")
          javac_path = nil unless File.exist?(javac_path)
       end
       javac_path ||= "javac" 
```

Reply:
@kou

Thanks for the review.  I'm not sure about your example, it seems a rather extreme and odd edge case.  Also, I'm not familiar with all of the *nix distributions.

`"/"` was a bad choice for default, as `File.join('/', 'bin', 'javac')"` and `File.join('', 'bin', 'javac')` both equal `"/bin/javac"`.

The previous code worked correctly when used with GHA.  But your changes also work.

Updated with your suggested code.

Reply:
Oh, sorry. I considered the following case:

* `ENV["PATH"]` is `"/usr/local/bin:/usr/bin`
* `ENV["JAVA_HOME"]` isn't set
* `/usr/local/bin/javac` exists
* `/usr/bin/javac` exists

In the case, `javac` uses `/usr/local/bin/javac` but `File.join(ENV.fetch("JAVA_HOME", "/"), "bin", "javac")` uses `/usr/bin/javac` not `/usr/local/bin/javac`.

Reply:
Sorry, I need to try this.

If `ENV["JAVA_HOME"]` isn't set, then `ENV.fetch("JAVA_HOME", "/"), "bin", "javac")` yields `/bin/javac`, I thought `File.exist?("/bin/javac")` would be false, and then `javac` would be used, and the file would be resolved by `PATH`?

Again, I need to try it...